frontend/src/App.test.tsx

```diff
@@ -62,9 +70,9 @@ test("data is fetched again after refresh interval", async () => {
 
   await waitFor(() => expect(fetch).toHaveBeenCalledTimes(1));
 
-  screen
-    .getAllByText("31 °C", { exact: false })
-    .forEach((temp) => expect(temp).toBeInTheDocument());
+  (await screen.findAllByText("31 °C", { exact: false })).forEach((temp) =>
+    expect(temp).toBeInTheDocument(),
+  );
   data[0].workStations[0].gpus[0].gpu_temp = 100;
```

Comment:
NIT: It might be advisable to deconstruct this line into semantically distinct variables (i.e. `workstation`, `gpu`, etc).

Reply:
Done